### PR TITLE
Fix GEM Offline DQM for GE2/1 Demonstrator (backport of #36883, 12_2_X)

### PR DESCRIPTION
**Pull request by seungjin-yang**
#### PR description:
Backport of #36883. This PR fixes MonitorElement: WARNING:setBinLabel caused by GEM offline DQM not handling the new GEM geometry with GE2/1 demonstrator. See https://github.com/cms-sw/cmssw/pull/36860#issuecomment-1028028804.

#### PR validation:
This PR is tested with 11634.911.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
This PR is a backport of #36883.

### CODE REVIEW

**Comment by perrotta** [comment]
Please test

**Comment by perrotta** [comment]
urgent

**Comment by jfernan2** [comment]
+1

File: DQMOffline/Muon/python/gemEfficiencyHarvester_cfi.py [hunk 1]
```diff
@@ -6,9 +6,9 @@
 from DQMOffline.Muon.gemEfficiencyAnalyzer_cfi import gemEfficiencyAnalyzerSta as _gemEfficiencyAnalyzerSta
```
Comment (perrotta) [inline DQMOffline/Muon/python/gemEfficiencyHarvester_cfi.py:7]: and the same for `gemEfficiencyHarvester`

**Review by perrotta — COMMENTED** [review]
I notice that there are two modules missing with respect to the corresponding version in the master:
- `gemEfficiencyAnalyzer`
- `gemEfficiencyHarvester`

Are they needed in 12_2, or not?
Could you please comment about the difference?

File: DQMOffline/Muon/python/gemEfficiencyAnalyzer_cfi.py [hunk 2]
```diff
@@ -27,30 +27,34 @@
```
[the remark this answers is not in the document]
Comment (seungjin-yang) [reply DQMOffline/Muon/python/gemEfficiencyAnalyzer_cfi.py:27]: `GEMEfficiencyAnalyzer::fillDescriptions` automatically generates `DQMOffline.Muon.gemEfficiencyAnalyzerDefault_cfi.py`. Then, `DQMOffline.Muon.gemEfficiencyAnalyzer_cfi` loads `GEMEfficiencyAnalyzer` module from  `DQMOffline.Muon.gemEfficiencyAnalyzerDefault_cfi`. Am I understanding your question correctly?

https://github.com/cms-sw/cmssw/blob/c24439ed1375b88eef7173a58c90ff71298547c2/DQMOffline/Muon/src/GEMEfficiencyAnalyzer.cc#L55-L117

File: DQMOffline/Muon/python/gemEfficiencyHarvester_cfi.py (hunk 1 above)
Comment (seungjin-yang) [reply DQMOffline/Muon/python/gemEfficiencyHarvester_cfi.py:7]: same with the previous question. I know the configuration files are not currently following the style guide, but I will fix it later in the master branch.

https://github.com/cms-sw/cmssw/blob/c24439ed1375b88eef7173a58c90ff71298547c2/DQMOffline/Muon/src/GEMEfficiencyHarvester.cc#L15-L19

**Comment by seungjin-yang** [comment]
> I notice that there are two modules missing with respect to the corresponding version in the master:
>
> - gemEfficiencyAnalyzer
> - gemEfficiencyHarvester
> Are they needed in 12_2, or not?

`GEMEfficiencyAnalyzer` and `GEMEfficiencyHarvester` are required for GEM offline DQM in 12_2X.  Both modules have been moved from `DQMOffline/Muon` to `DQM/GEM` package in the master branch for better packaging.

**Comment by perrotta** [comment]
Ok, thank you @seungjin-yang , [that](https://github.com/cms-sw/cmssw/pull/36906#issuecomment-1032705533) makes sense (and indeed the python configuration in this backport PR do correspond to what you write)

**Pull request merged**